x-pack/legacy/plugins/alerting/server/plugin.ts

```diff
@@ -128,11 +128,12 @@ export class Plugin {
 
     this.taskRunnerFactory.initialize({
       logger: this.logger,
-      getServices(request: Hapi.Request): Services {
+      getServices(rawRequest: Hapi.Request): Services {
+        const request = KibanaRequest.from(rawRequest);
         return {
-          callCluster: (...args) =>
-            adminClient!.asScoped(KibanaRequest.from(request)).callAsCurrentUser(...args),
-          savedObjectsClient: core.savedObjects.getScopedSavedObjectsClient(request),
+          callCluster: (...args) => adminClient!.asScoped(request).callAsCurrentUser(...args),
+          // rawRequest is actually a fake request, converting it to KibanaRequest causes issue in SO access
+          savedObjectsClient: core.savedObjects.getScopedSavedObjectsClient(rawRequest as any),
```

Comment:
I did properly adapt this to use `core.savedObjects.getScopedSavedObjectsClient(request)`. However I got test failures, and I isolated it to this change.

Calling `core.savedObjects.getScopedSavedObjectsClient` with `KibanaRequest.from(request)` instead of the raw HAPI request makes the test fail, with 
`Saved object [alert/dbbfe2a1-e3ab-4ebe-84f2-9ed495b9d81b] not found`

Forcecasting the raw request fix the failure:

```typescript
savedObjectsClient: core.savedObjects.getScopedSavedObjectsClient(rawRequest as any)
```

AFAIK it's because the `rawRequest`, even if typed as `Hapi.Request`, is actually a fake request:

https://github.com/elastic/kibana/blob/ea9a7b8a16abbee97ddf39f91e4cf3c3b7253b42/x-pack/legacy/plugins/alerting/server/task_runner/task_runner.ts#L67-L82

and the `GetServices` signature differs from between the type and the actual implementation, allowing to call it without a proper hapi request...

https://github.com/elastic/kibana/blob/5853360d75ee17cd00c777d2b2c28fd2f8abd727/x-pack/legacy/plugins/alerting/server/types.ts#L15

Not sure what the underlying cause is though. I though only the headers from the request were actually used when using the request to create the scoped repository. But the fake request does miss some properties used when converting from request to kibanaRequest such as `method`.

Is this comment acceptable/enough for now, or should I investigate in this PR?

Reply:
This is related to https://github.com/elastic/kibana/issues/39430 btw

Reply:
> Is this comment acceptable/enough for now, or should I investigate in this PR?

As for me, it's not a blocker for the current PR. But we should prioritize https://github.com/elastic/kibana/issues/39430 to be able to cut off legacy server-side plugins in 7.7